zepid/causal/doublyrobust/TMLE.py

```diff
@@ -77,35 +77,42 @@ def exposure_model(self, model, custom_model=None, bound=False, print_results=Tr
-            if bound is not False:  # Bounding predicted probabilities if requested
-                self.gW = self._bounding(self.gW, bounds=bound)
+            self.g1W = fitmodel.predict(self.df)
 
         # User-specified prediction model
         else:
-            try:  # This two-stage 'try' filters whether the data needs an intercept, then has the predict() attr
-                data = patsy.dmatrix(model, self.df)
-                try:
-                    self.gW = custom_model.predict(data)
-                    if bound is not False:  # Bounding predicted probabilities if requested
-                        self.gW = self._bounding(self.gW, bounds=bound)
-                except AttributeError:
-                    raise AttributeError("custom_model does not have the 'predict()' attribute")
-            except ValueError:
-                data = patsy.dmatrix(model+' - 1', self.df)
-                try:
-                    if hasattr(custom_model, 'predict_proba'):
-                        self.gW = custom_model.predict_proba(data)[:, 1]
+            if hasattr(custom_model, 'fit'):
+                # Supports sklearn and supylearner
+                if {'X', 'y'}.issubset(set(inspect.getfullargspec(custom_model.fit)[0])):
+                    data = patsy.dmatrix(model + ' - 1', self.df)
+                    fm = custom_model.fit(X=data, y=self.df[self._outcome])
+                    if hasattr(fm, 'predict_proba'):
+                        self.g1W = fm.predict_proba(data)[:, 1]
+                    elif hasattr(fm, 'predict'):
+                        self.g1W = fm.predict(data)
                     else:
-                        self.gW = custom_model.predict(data)
-                    if bound is not False:  # Bounding predicted probabilities if requested
-                        self.gW = self._bounding(self.gW, bounds=bound)
-                except AttributeError:
-                    raise AttributeError("custom_model does not have the 'predict()' or 'predict_proba()' attribute")
-        # Setting flag for fit() check to make sure exposure model was specified
+                        raise ValueError("Currently custom_model must have 'predict' or 'predict_proba' attribute")
+                    if print_results and hasattr(fm, 'summarize'):
+                        fm.summarize()
+
+                else:
+                    raise ValueError("Currently custom_model must have the arguments [X, y]. This "
+                                     "covers sklearn and supylearner. If there is a predictive model you would "
+                                     "like to use, please open an issue at https://github.com/pzivich/zepid and I "
+                                     "can work on adding support")
+
+            else:
+                raise AttributeError("Custom model does not have the 'fit' method")
+
+        self.g0W = 1 - self.g1W
+        if bound is not False:  # Bounding predicted probabilities if requested
```

Comment:
`if not bound`?

Reply:
I can shorten this to ``if bound``. ``not bound`` puts unbounded into this loop.